cairo-contracts/protobuf/src/types/message.cairo

```diff
@@ -0,0 +1,140 @@
+use protobuf::types::tag::{WireType, ProtobufTag, ProtobufTagImpl};
+use protobuf::primitives::numeric::NumberAsProtoMessage;
+use protobuf::types::wkt::Any;
+
+pub trait ProtoMessage<T> {
+    fn decode_raw(ref value: T, serialized: @ByteArray, ref index: usize, length: usize);
+    fn encode_raw(self: @T, ref output: ByteArray);
+    fn wire_type() -> WireType;
+    fn type_url() -> ByteArray;
```

Comment:
I think the `type_url` method should be placed in a separate trait, as not all type implements it.

Reply:
Good point. I see that prost has two traits `Name` and `Message` with `Name: Message` bound to avoid this.